Only in drivers/w1/: patch
diff -up drivers/w1/w1_netlink.c drivers/w1/patch/w1_netlink.c
--- drivers/w1/w1_netlink.c	2015-02-07 11:04:31.000000000 +0100
+++ drivers/w1/patch/w1_netlink.c	2015-02-09 21:56:46.000000000 +0100
@@ -321,6 +321,13 @@ static int w1_process_command_io(struct
 	case W1_CMD_WRITE:
 		w1_write_block(dev, cmd->data, cmd->len);
 		break;
+	case W1_CMD_WRITE_PULLUP:
+		if (cmd->len>=sizeof(u16)) {
+			w1_next_pullup(dev, *(u16*)(cmd->data));
+			w1_write_block(dev, &cmd->data[sizeof(u16)], cmd->len-sizeof(u16));
+		} else
+			err = -EINVAL;
+		break;
 	default:
 		err = -EINVAL;
 		break;
@@ -381,6 +388,7 @@ static int w1_process_command_master(str
 		break;
 	case W1_CMD_READ:
 	case W1_CMD_WRITE:
+	case W1_CMD_WRITE_PULLUP:
 	case W1_CMD_TOUCH:
 		err = w1_process_command_io(dev, req_cmd);
 		break;
diff -up drivers/w1/w1_netlink.h drivers/w1/patch/w1_netlink.h
--- drivers/w1/w1_netlink.h	2015-02-07 11:04:31.000000000 +0100
+++ drivers/w1/patch/w1_netlink.h	2015-02-09 21:22:57.000000000 +0100
@@ -115,7 +115,9 @@ enum w1_commands {
 	W1_CMD_SLAVE_ADD,
 	W1_CMD_SLAVE_REMOVE,
 	W1_CMD_LIST_SLAVES,
-	W1_CMD_MAX
+	W1_CMD_MAX,
+
+	W1_CMD_WRITE_PULLUP=20
 };
 
 /**
